test/lib/helpers/doChannelReportTest.js

```diff
@@ -0,0 +1,69 @@
+'use strict';
+
+const assert          = require('assert');
+const sinon           = require('sinon');
+const mockBot         = require('../../support/mockBot');
+const dbHelper        = require('../../support/persistenceHelper');
+
+const doChannelReport = require('../../../lib/helpers/doChannelReport');
+
+describe('doChannelReport', () => {
+  let bot, channelName, userName;
+  let state;
+
+  beforeEach(() => {
+    bot         = mockBot();
+    channelName = 'channelName';
+    userName    = 'slackUserName';
+  });
+
+  describe('when all the records are present, and the bot works', () => {
+    beforeEach((done) => {
+      state = {};
+      dbHelper
+        .setupTest(channelName, state)
+        .then(() => { done(); });
+    });
+
+    describe('when updating', () => {
+      it('updates the report', (done) => {
+        let spy = sinon.spy();
+        bot.api.chat.update = spy;
+        doChannelReport(bot, channelName, true, userName, () => {
+          assert(bot.api.chat.update.calledOnce);
+          let report = spy.args[0][0];
+          let callback = spy.args[0][1];
+          assert(report.attachments.match(/Todays standup for <#channelName>/));
+          callback(); // just to make sure no errors, can't make assertions
+          // because it does not use the callback argument
```

Comment:
If we mock the database, would we also want to verify that it was called?  And if not, should we be checking what's in the database after this?  That would make this `callback()` meaningful, since those things might not happen until inside the callback.